camayoc/tests/qcs/test_profiles.py

```diff
@@ -149,26 +244,59 @@ def test_read_all(cleanup):
            created network profiles
         3) Confirm that all profiles are in the list.
     :expectedresults: All profiles are present in data returned by API.
-    :caseautomation: notautomated
+    :caseautomation: automated
     """
-    pass
+    created_profs = []
+    for i in range(random.randint(2, 20)):
+        # gen_valid_profs will take care of cleanup
+        created_profs.append(
+            gen_valid_profile(
+                cleanup,
+                shared_client,
+                'localhost'))
+    server_profs = NetworkProfile().list().json()
+    for sp in server_profs:
+        found = False
+        for i, cp in enumerate(created_profs):
+            if sp['id'] == cp.fields()['id']:
+                if cp.equivalent(sp):
+                    found = True
+                    break
+        if found:
+            created_profs.pop(i)
+    # if we found everything we created, then the list should be empty
+    assert len(created_profs) == 0
 
 
-@pytest.mark.skip
-def test_delete(cleanup):
+def test_delete(cleanup, shared_client):
     """After creating several network profiles, delete one.
 
     :id: 24d811b1-655d-4278-ab9f-64ca46a7121b
     :description: Test that we can delete an individual network profile by id
     :steps:
         1) Create collection of network profiles, saving the information.
         2) Send a DELETE request to destroy individual profile
-        3) Send GET request to network profile endpoint to get list of
-           created profiles.
-        4) Confirm that all profiles are in the list except the deleted one.
-        5) Repeat until all profiles are deleted.
-    :expectedresults: All profiles are present in data returned by API except
-        the deleted profle.
-    :caseautomation: notautomated
+        3) Confirm that all profiles are on the server except the deleted one.
+        4) Repeat until all profiles are deleted.
+    :expectedresults: All profiles are present on the server except the
+        deleted profile.
+    :caseautomation: automated
     """
-    pass
+    created_profs = []
+    num_profs = random.randint(2, 20)
+    echo_client = api.Client(response_handler=api.echo_handler)
+    for i in range(num_profs):
+        # gen_valid_profs will take care of cleanup
+        created_profs.append(
+            gen_valid_profile(
+                cleanup,
+                shared_client,
+                'localhost'))
+    for i in range(num_profs):
```

Comment:
When the iteration variable is not used it is better to name it `_` this way the code tells that we need the loop but the iteration variable is not really needed.